src/scalar/dusk.rs

```diff
@@ -253,6 +254,35 @@ impl Scalar {
         res
     }
 
+    /// Compute a uniformly distributed random scalar.
+    ///
+    /// Because scalars take 255 bits for encoding it is difficult to generate
+    /// random bit-pattern that ensures to encodes a valid scalar.
+    /// Wrapping the values that are higher than [`MODULUS`] results in hitting
+    /// some values more than others, and zeroing out the highest two bits will
+    /// eliminate some values from the possible results.
+    ///
+    /// This function achieves a uniform distribution of scalars by using
+    /// rejection sampling: random bit-patterns are generated until a valid
+    /// scalar is found.
+    /// The function is not constant time but that shouldn't be a concern since
+    /// no information about the scalar can be gained by knowing the time of
+    /// its generation.
+    pub fn uni_random(rng: &mut (impl RngCore + CryptoRng)) -> Self {
+        let mut buf = [0; 32];
+        let mut scalar: Option<Self> = None;
+
+        // we loop as long as it takes to generate a valid scalar
+        while scalar == None {
```

Comment:
I don't like the infinite loops (even if this would not happen for valid randomizers)

Does make sense to have a max try count?

Reply:
I understand your point, will implement a un-failable fallback strategy

Reply:
Max try count is not a good idea. All of a sudden you're not generating uniform random numbers anymore, and the user won't even know.

Reply:
Indeed, to me it's ok to return None and let the user handle it (without any masked fallback procedure)

But I'm not strongly opinionated

Reply:
I'm not sure why we would do that in the first place. Is there any reason why we wouldn't put that infinite loop there?

Reply:
To be clear, I don't like infinite loops either, but this is not a real "infinite loop". It has a an exit condition that will always be reached eventually.

Reply:
I'm ok for the infinite loop if it's properly documented.

Mine was just a nit

Reply:
I would really like to avoid returning an `Option` as this adds unnecessary friction downstream.
I do not mind setting a max-try count or looping until a valid number is found since with the correct implementation of rng we should never reach any set max limit anyway. When we set the msb to 0, we have a change of less than 0.5 of hitting an invalid scalar, so with a max limit of for example 42, we have a chance of less than 
P_invalid < 2.27373675443e-13 ( = 0.5^42 )
of not hitting a correct scalar. This is a negligible probability and the functionality should not be different with either of the strategies we chose